specifyweb/specify/tree_views.py

```diff
@@ -343,4 +351,392 @@ def perm_target(tree):
+    'lithostrat': (LITHO_STRAT_RANKS, LITHO_STRAT_INCREMENT),
+}
+
+# Wrapper for tree editing api functions
+def tree_edit(edit_func):
+    @login_maybe_required
+    @require_POST
+    @transaction.atomic
+    @wraps(edit_func)
+    def wrapper(*args, **kwargs):
+        try:
+            edit_func(*args, **kwargs)
+            result = {'success': True}
+        except BusinessRuleException as e:
+            result = {'success': False, 'error': str(e)}
+            return HttpResponseServerError(toJson(result), content_type="application/json")
+        return HttpResponse(toJson(result), content_type="application/json")
+
+    return wrapper
+
+@openapi(schema={
+    'post': {
+        "requestBody": {
+            "required": True,
+            "description": "Add rank to a tree.",
+            "content": {
+                "application/json": {
+                    "schema": {
+                        "type": "object",
+                        "description": "The request body.",
+                        "properties": {
+                            "newRankName": {
+                                "type": "string",
+                                "description": "The name of the new rank to add."
+                            },
+                            "parentRankName": {
+                                "type": "string",
+                                "description": "The name of the parent rank to add the new rank to (use 'root' to add to the front)."
+                            },
+                            "treeName": {
+                                "type": "string",
+                                "description": "The name of tree."
+                            },
+                            "treeID": {
+                                "type": "integer",
+                                "description": "The ID of the tree (defaults to the first tree)."
+                            },
+                            "newRankTitle": {
+                                "type": "string",
+                                "description": "The title of the rank to add (defaults to the name)."
+                            },
+                            "useDefaultRankIDs": {
+                                "type": "boolean",
+                                "description": "Determine if the default rank IDs should be used (defaults to True)."
+                            }
+                        },
+                        'required': ['newRankName', 'parentRankName'],
+                        'additionalProperties': False
+                    }
+                }
+            }
+        },
+        "responses": {
+            "200": {"description": "Success"},
+            "400": {"description": "Cannot add tree rank with those parameters"},
+            "500": {"description": "Server Error"}
+        }
+    },
+})
+@login_maybe_required
+@require_POST
+@transaction.atomic
+def add_tree_rank(request, tree) -> HttpResponse:
```

Comment:
Could someone take a brief look at what magic sp6 might be doing when it adds a tree rank just to make sure we are not missing any business logic here?